source/parser.hera

```diff
@@ -2921,22 +2933,31 @@ IfStatement
   # NOTE: Added paren-less condition
   # NOTE: Block isn't Statement so we can handle implied braces by nesting
   (IfClause / UnlessClause):clause Block:block ElseClause?:e ->
-    const children = [...clause.children, block]
+    const children = [...clause.children]
+    block = blockWithPrefix(clause.condition.expression, block)
+    children.push(block)
     if (block.bare && e) children./**/push(";")
     if (e) children./**/push(e)
+
     return {
       type: "IfStatement",
       then: block,
       else: e,
       children,
+      condition: clause.condition,
```

Comment:
Would it make sense to instead start this object literal with `...clause`? Then could also omit `type` here.